extensions/modules/file/src/test/resources/util/helper.xqm

```diff
@@ -0,0 +1,192 @@
+(:
+ : eXist-db Open Source Native XML Database
+ : Copyright (C) 2001 The eXist-db Authors
+ :
+ : info@exist-db.org
+ : http://www.exist-db.org
+ :
+ : This library is free software; you can redistribute it and/or
+ : modify it under the terms of the GNU Lesser General Public
+ : License as published by the Free Software Foundation; either
+ : version 2.1 of the License, or (at your option) any later version.
+ :
+ : This library is distributed in the hope that it will be useful,
+ : but WITHOUT ANY WARRANTY; without even the implied warranty of
+ : MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ : Lesser General Public License for more details.
+ :
+ : You should have received a copy of the GNU Lesser General Public
+ : License along with this library; if not, write to the Free Software
+ : Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ :)
+xquery version "3.1";
+
+module namespace helper="http://exist-db.org/xquery/test/util/helper";
+import module namespace fixtures="http://exist-db.org/xquery/test/util/fixtures" at "fixtures.xqm";
+import module namespace file="http://exist-db.org/xquery/file";
+import module namespace xmldb="http://exist-db.org/xquery/xmldb";
+import module namespace util="http://exist-db.org/xquery/util";
+
+declare variable $helper:error := xs:QName("helper:assert-sync-error");
+
+(:
+/db
+    /file-module-test
+        /data
+            test-data.xml
+        test-text.txt
+        test-query.xq
+        bin
+:)
+declare function helper:setup-db() as empty-sequence() {
+    (
+        xmldb:create-collection("/db", $fixtures:collection-name),
+        helper:create-db-resource($fixtures:collection, "test-text.txt", $fixtures:TXT),
+        helper:create-db-resource($fixtures:collection, "test-query.xq", $fixtures:XQY),
+        helper:create-db-resource($fixtures:collection, "bin", $fixtures:BIN),
+
+        xmldb:create-collection($fixtures:collection, $fixtures:child-collection-name),
+        helper:create-db-resource($fixtures:child-collection, "test-data.xml", $fixtures:XML)
+    )
+    => helper:to-empty-sequence()
+};
+
+declare function helper:clear-db() {
+    xmldb:remove($fixtures:collection)
+};
+
+declare function helper:create-db-resource($collection as xs:string, $resource as xs:string, $content as item()) as empty-sequence() {
+    (
+        xmldb:store($collection, $resource, $content),
+        xmldb:touch($collection, $resource, $fixtures:mod-date)
+    )
+    => helper:to-empty-sequence()
+};
+
+declare function helper:modify-db-resource($collection as xs:string, $resource as xs:string) as empty-sequence() {
+    xmldb:touch($collection, $resource, $fixtures:mod-date-2)
+    => helper:to-empty-sequence()
+};
+
+declare function helper:clear-suite-fs ($suite as xs:string) as empty-sequence() {
+    helper:glue-path((
+        util:system-property("java.io.tmpdir"),
+        $suite
+    ))
+    => helper:clear-fs()
```

Comment:
This should be refactored to `return helper:clear-fs()` so that the test is easy to debug